File: gamedbd/src/main/java/bgame/db/config/server/DBServerConfig.java
package bgame.db.config.server;

import java.util.List;

import bestan.common.logic.ServerConfig;
import bestan.common.lua.BaseLuaConfig;
import bestan.common.lua.LuaParamAnnotation;
import bestan.common.lua.LuaParamAnnotation.LuaParamPolicy;
import bestan.common.net.server.NetServerConfig;
import bestan.common.thread.BExecutor;

public class DBServerConfig extends BaseLuaConfig {
	/**
	 * 通用配置
	 */
	public ServerConfig serverConfig;
	/**
	 * netServer配置
	 */
	public NetServerConfig netServerCfg;
	/**
	 * 线程池数量
	 */
	public int nThreadPool;
	/**
	 * timer更新间隔（毫秒）
	 */
	public int timerIickInterval;
	/**
	 * message包列表
	 */
	public List<String> messagePackages;
	/**
	 * messageHandler包列表
	 */
	public List<String> messageHandlerPackages;
	
	/**
	 * 工作线程运行时设置
	 */
	@LuaParamAnnotation(policy=LuaParamPolicy.OPTIONAL)
	public BExecutor workExecutor;
	
	public static DBServerConfig instance;
}

File: server-common/server_common_res/server_config/DBServerConfig.lua
--1kb
local KB = 1024
--1兆
local MB = KB * KB

local serverConfig =
{
	zoneId = 1001,
	
	tickInterval = 1000,
	
	managerType = 0,
	
	playerTickInterval = 100,
	
	messagePackage = "bgame.common.message",

	messageHandlerPackage = "",	
}

local netServerCfg=
{
	serverName = "DBServer",
	bossGroupThreadCount = 1,
	workerGroupThreadCount = 1,
	serverIP = "127.0.0.1",
	serverPort = 1019,
	--serverchanel(用来监听和接受连接)的接收缓冲区大小
	optionRcvbuf = 64 * KB,
	--serverchanel(用来监听和接受连接)的发送缓冲区大小
	optionSndbuf = 128 * KB,
	--clientchannel（每一个建立连接）的接收缓冲区大小
	childOptionRcvbuf = 16 * KB,
	--clientchannel（每一个建立连接）的发送缓冲区大小
	childOptionSndbuf = 32 * KB,
}

local config = 
{
	serverConfig = serverConfig,
	netServerCfg = netServerCfg,
	
	--工作线程池数量
	nThreadPool = 10,
	
	--timer更新间隔（毫秒）
	timerIickInterval = 500,
	
	--消息包列表
	messagePackages = { "bgame.common.message" },
	
	--消息处理包列表
	messageHandlerPackages = { "bgame.db.net.handler", "bestan.common.db.handler" },
}

return config